crates/shrs_core/src/plugin.rs

```diff
@@ -12,6 +12,14 @@ pub struct PluginMeta {
     pub name: String,
     pub description: String,
 }
+impl PluginMeta {
+    pub fn new(name: &str, description: &str) -> Self {
```

Comment:
Can make this
```rust
pub fn new<S: ToString>(name: S, description: S)
```
to support passing both `String` and `&str`

Reply:
Actually I would say that this constructor is not needed since all the fields are public anyways. It would be more self documenting to initialize the `PluginMeta` struct directly

Reply:
I thought it was annoying to type .to_string everwhere